pennylane/labs/vibrational_ham/bosonic.py

```diff
@@ -0,0 +1,538 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""The bosonic representation classes and functions."""
+from copy import copy
+
+import pennylane as qml
+from pennylane.typing import TensorLike
+
+# pylint: disable= too-many-nested-blocks, too-many-branches, invalid-name
+
+
+class BoseWord(dict):
+    r"""Immutable dictionary used to represent a Bose word, a product of bosonic creation and
+    annihilation operators, that can be constructed from a standard dictionary.
+
+    The keys of the dictionary are tuples of two integers. The first integer represents the
+    position of the creation/annihilation operator in the Bose word and the second integer
+    represents the orbital it acts on. The values of the dictionary are one of ``'+'`` or ``'-'``
+    symbols that denote creation and annihilation operators, respectively. The operator
+    :math:`a^{\dagger}_0 a_1` can then be constructed as
+
+    >>> w = BoseWord({(0, 0) : '+', (1, 1) : '-'})
+    >>> w
+    b⁺(0) b(1)
+    """
+
+    # override the arithmetic dunder methods for numpy arrays so that the
+    # methods defined on this class are used instead
+    # (i.e. ensure `np.array + BoseWord` uses `BoseWord.__radd__` instead of `np.array.__add__`)
+    __numpy_ufunc__ = None
+    __array_ufunc__ = None
+
+    def __init__(self, operator, christiansen_boson=False):
```

Comment:
Can we use a simpler name instead of `christiansen_boson`? Something like `cform = False`?